engine/syntax_checker.ts

```diff
@@ -0,0 +1,20 @@
+import * as fs from 'fs';
+import path = require('path');
+
+export class SyntaxChecker {
```

Comment:
If I understand it right the SyntaxChecker just prints the syntax errors to a file, maybe you choose than a better self-describing name like SyntaxErrorLogger.